sdks/go/pkg/beam/io/xlang/schemaio/schemaio_test.go

```diff
@@ -0,0 +1,80 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schemaio
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/core/runtime/xlangx"
+	"github.com/google/go-cmp/cmp"
+)
+
+func init() {
+	beam.RegisterType(reflect.TypeOf((*testConfig)(nil)))
+	beam.RegisterType(reflect.TypeOf((*testDataSchema)(nil)))
+}
+
+type testConfig struct {
+	cfg1 string
+	cfg2 *int
+}
+
+type testDataSchema struct {
+	scm1 bool
+	scm2 float64
```

Comment:
Please Export Fields. Otherwise things are likely not working as well as one might expect. Unexported fields are simply not encoded.

Reply:
Done.